node/src/components/linear_chain_sync/operations.rs

```diff
@@ -421,5 +561,29 @@ where
         outstanding_trie_keys.extend(missing_descendant_trie_keys);
     }
 
-    Ok(())
+    while let Some(block_with_metadata) = fetch_and_store_block_by_height(
```

Comment:
Should we add another condition to exit the loop? Whenever we have a reasonably recent block, e.g. one that is in the current era?

Reply:
This is issue #1549, but @fizyk20 and I just discussed this and I could tackle this here...

Reply:
Ah, right, I didn't know there's an issue for it. :+1: